File: LeetCode/122. Best Time to Buy and Sell Stock II.py
# 同样单次遍历可以解决，只要后一天价格比前一天高，就认为可以前一天卖后一天卖，多天的买卖等价于这段时间每一天的和

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        for index, price in enumerate(prices):
            if (index == 0):
                continue
            
            profit = price - prices[index - 1]
            if (profit > 0):
                max_profit += profit
            
        
        return max_profit

File: LeetCode/122. Best Time to Buy and Sell Stock II.cpp
// 这个思路也很简单，每次一个价格比前面高就加起来就行
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() < 1) {
            return 0;
        }
        int res = 0;
        // int left = prices[0];
        for (int i = 1; i < prices.size(); ++i) {
            if (prices[i] > prices[i - 1]) {
                res += prices[i] - prices[i - 1];
            }
        }
        return res;
    }
};